requirements.txt

```diff
@@ -8,7 +8,7 @@ wasabi>=0.8.1,<1.1.0
 catalogue>=2.0.4,<2.1.0
 ml_datasets>=0.2.0,<0.3.0
 # Third-party dependencies
-pydantic>=1.7.4,!=1.8,!=1.8.1,<1.9.0
+pydantic>=1.9.0,<1.10.0
```

Comment:
Is there a particular reason not to support the older versions of `pydantic` anymore?

Reply:
The older versions of `pydantic` aren't compatible with newer versions of `mypy`, and it didn't seem worth trying to specify these complex constraints in `setup.cfg` (if it's even possible).

Reply:
Hmm, let me think about this again. There is no way to express complex constraints like that.

However, `mypy` is always an optional dependency, so I think it might be better to have a wider `pydantic` constraint so that `thinc` remains more compatible with other packages.

What might be possible is to let `requirements.txt` diverge from `setup.cfg` to reflect the constraints for the development/CI environment that do require newer versions of `pydantic` for the specified version of `mypy`, but I'm not entirely sure we'd want to do that.

Reply:
Actually, I think we can just leave it as the wider range in `requirements.txt` and `mypy` itself will constraint `pydantic` when you use `requirements.txt`.

Reply:
No, I think I was confused :-). There's no need for a way to express complex constraints like that because they're already expressed by the transitive setup configurations of the dependencies you specify: I think you're right that keeping the wider range is the best policy.